src/java.base/share/classes/java/lang/reflect/code/op/ExtendedOps.java

```diff
@@ -3145,46 +3142,66 @@ public static PatternOps.RecordPatternOp recordPattern(RecordTypeDesc recordDesc
     }
 
     @OpDeclaration(StringTemplateOp.NAME)
-    public static final class StringTemplateOp extends OpWithDefinition {
+    public static final class StringTemplateOp extends OpWithDefinition implements Op.Nested {
 
         public static final String NAME = "java.stringTemplate";
 
+        private final Value processorValue;
+        private final List<Value> literalsValues;
```

Comment:
These two fields are redundant, as we can always refer to the operands directly.

Probably better to refer to the literals as fragments like they are referred in the compiler and i presume the language specification.

If you wish add methods called `processor` and `fragments` that return the operands.